### PR TITLE
Fix for incorrect checking of error

### PR DESCRIPTION
**Pull request by kirankumar-ravuri**
There is issue with if condition checking for address_in_use case.

### CODE REVIEW

File: source/core/owt_base/RawTransport.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ void RawTransport<prot>::listenTo(uint32_t minPort, uint32_t maxPort)
                 m_socket.tcp.acceptor->open(tcp::v4());
                 m_socket.tcp.acceptor->bind(tcp::endpoint(tcp::v4(), port), ec);
 
-                if (ec != boost::system::errc::address_in_use) {
```
Comment (starwarfan) [inline source/core/owt_base/RawTransport.cpp:228]: If "ec" is "address_in_use" here, the loop will continue until successfully finding a port to bind.
Other "ec" conditions will be handled in line 228.
In which situation you want it to break in (!ec)?

Comment (kirankumar-ravuri) [reply source/core/owt_base/RawTransport.cpp:228]: I think ec will be zero for success, all other error cases will return non zero value. Isn't the loop supposed to continue on error and come out of the loop as soon as there is a success or until the ports are exhausted?

Comment (starwarfan) [reply source/core/owt_base/RawTransport.cpp:228]: The loop is supposed to continue on address_in_use error, will break on success or other error.

Comment (kirankumar-ravuri) [reply source/core/owt_base/RawTransport.cpp:228]: OK. I got the intention.

I tried to include some logs in the code, like below.  I am using 1.53.0 for boost. 

```
ELOG_WARN("TCP transport - checking port number : %d", port);
ELOG_WARN("TCP transport returned ec: %s, %d", ec.message().c_str(), ec.value());
if (ec != boost::system::errc::address_in_use) {
    ELOG_WARN("Break");
    break;
} 
```
Below are some of the logs from the modified code.

> 2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport - checking port number : 30003
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport returned ec: Address already in use, 98
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - Break
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport listen in port range bind error: Address already in use
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - TCP transport - checking port number : 30000
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - TCP transport returned ec: Success, 0
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - Break

`if (ec != boost::system::errc::address_in_use)` condition doesn't seem to work always. 

Is there any dependency on boost version ?

Comment (starwarfan) [reply source/core/owt_base/RawTransport.cpp:228]: The error code log was "TCP transport returned ec: Address already in use, 98", but the "if (ec != boost::system::errc::address_in_use)" condition succeed? Did this happen always?

Comment (kirankumar-ravuri) [reply source/core/owt_base/RawTransport.cpp:228]: I think if condition should be replaced with this instead.
 `if(ec.value() != boost::system::errc::address_in_use)`

Comment (starwarfan) [reply source/core/owt_base/RawTransport.cpp:228]: Hi, thanks for your suggestion.
Can you update your code in pull request so we can merge it?

Comment (kirankumar-ravuri) [reply source/core/owt_base/RawTransport.cpp:228]: @starwarfan updated the request.

**Pull request merged**